src/snowflake/snowpark/session.py

```diff
@@ -333,6 +341,11 @@ def create(self) -> "Session":
                 _add_session(session)
             else:
                 session = self._create_internal(self._options.get("connection"))
+
+            if self._app_name:
+                app_name_tag = f"APPNAME={self._app_name}"
+                session.query_tag = app_name_tag
+
```

Comment:
this will override the query tags. Is that the intended behavior? Can you use append query tag using a newer API added recently

Reply:
This is used in the `SessionBuilder.create` method which always returns a new session. Or are we injecting query_tags somewhere else?

Reply:
It is  `Session.append_query_tag`

Reply:
The query tag could be set at the user level so even if the session is fresh there could be an existing query_tag.

Reply:
changed to `Session.append_query_tag`